.github/workflows/run-eslint-check.yml

```diff
@@ -0,0 +1,23 @@
+name: Run ESlint check
+
+on:
+  push:
+    branches:
+      - '*'
+    tags:
+      - 'v*'
```

Comment:
Can we setup the following flow:

On PR open/change:

- ESLint check
- Build check

On push to `main`:

- ESLint check
- Build check
- Deploy API

Reply:
The workflow now goes as follows:

On PR open/change and all pushes:
    ESLint check
    Build check
    Run tests

On push to main:
    ESLint check
    Build check
    Run tests
    Deploy API